### DISCUSSION TITLE
Setting the charge hour does not work

### DISCUSSION THREAD

**Discussion by calliou1234** [flobz/psa_car_controller#196] [Q&A] [answered]
Hello,

the command

`http://192.168.178.35:4995/charge_hour?vin=VR3XXXXXXXXXXXX&hour=22&min=30`

brings an error.

Is there anything wrong with my command?

I've tried several times now but it doesn't work.

Thanks for your help
Thomas

**Comment by HOzo123** [comment]
Hello,

I have the same concern: using the function "change the charge hour" with command:
"http://127.0.0.1:5000/charge_hour?vin=VIN_VEHICLE&hour=2&min=31"
Remark: I have tried to change hour from 2h30 to 2h31

and I have the feedback:
"Bad Request
The browser (or proxy) sent a request that this server could not understand." 

I have performed a minor modification in the Python code on the file "views.py" as:
- Before:
@app.route('/charge_hour')
def change_charge_hour():
    return jsonify(CONFIG.myp.change_charge_hour(request.form['vin'], request.form['hour'], request.form['minute']))

- After 
@app.route('/change_charge_hour/<string:vin>/<int:hour>/<int:minute>')
def change_charge_hour(vin, hour, minute):
    return jsonify(CONFIG.myp.change_charge_hour(vin, hour, minute))

now, I can update the "delayed hour" (and I hope without any impact) using the command:
"/change_charge_hour/VIN/Heure/Minute"

However, it's probably not the best way to solve this concern because the function seems working perfectly for many users... there is probably a concern with "my" configuration but I'm not yet able to confirm what's the root cause.

Note: I'm using the last code from @flobz, on a Raspberry-Debian

H_Ozo

**Comment by flobz** [comment, marked as answer]
it's fixed.

**Reply by Aelbaert** [reply]
Sorry, but I get the same feedback: "Bad Request
The browser (or proxy) sent a request that this server could not understand."

I'm using v3.0.6 (in Home Assistant) on a Raspberry Pi 4 

Any suggestions how to proceed?

**Reply by flobz** [reply]
Open an issue with log please.

**Reply by Aelbaert** [reply]
Sorry Florian, not sure where to get the correct logs. I've added the Home Assistant - PSA Car controller logs for the last few minutes. The commands that give the error 'Bad Request
The browser (or proxy) sent a request that this server could not understand.' are coloured red in my screen.
The test-command:  'http://192.168.68.135:5000/charge_hour?vin=VXKUHZKXZL4386524&hour=23&min=12)'
If more info needed, please point me to location where to get please.
Regards and thanks for your quick response!, Albert
[HA-PSA-log.txt](https://github.com/flobz/psa_car_controller/files/10032037/HA-PSA-log.txt)
